rfcs/025-lineage-stage-0.md

```diff
@@ -0,0 +1,153 @@
+The first iteration is meant to be a fast implementation of the feature that we can get feedback on and improve in future iterations.
+
+## Motivation
+
+Lineage is essential to improving data discovery in Amundsen because it allows users to know where the data for a given resource is coming from as well as where this data is used downstream. 
+
+
+## Guide-level Explanation (aka Product Details)
+
+### New Concepts
+- Lineage: Lineage is a term that describes the flow of data from one entity to another. While this term can broadly include everything from services, events, ETLs, and dashboards, we will focus on table-to-table and column-to-column data lineage in this RFC. 
+- Upstream: Upstream is a relative term that describes data sources from which we inherit. Data flows from upstream to downstream.
+- Downstream: Downstream is a relative term that describes data entities which consume our data. 
+
+This feature will expose upstream and downstream tables and columns within the `Table Details` page.
+
+Those implementing Amundsen should keep in mind that this feature is meant to provide them with a way to surface their existing lineage data by calling the service containing that data from the metadata service. This iteration won't provide a model to persist lineage on neo4j, but rather a gateway to lineage data so it can be included on lineage API responses to displayed in frontend. It is also important to understand that the feature will be disabled by default and can be enable through configuration.
+
+
+## UI/UX-level Explanation
+
+![Table Lineage Preview](assets/025/table-lineage-preview.png)
+
+We will add two additional tabs to the `Table Details` page, `Upstream` and `Downstream`. Each tab will contain a list of tables from which data is inherited or consumed. This allows users view a table's lineage in a very simple manner.
+
+![Column Lineage Preview](assets/025/column-lineage-preview.png)
+
+Additionally we will add lineage information at the column level, viewable by expanding column metadata.
+
+These features will only appear when the lineage feature is enabled.
+
+## Reference-level Explanation (aka Technical Details)
+### Architecture
+
+![Lineage Stage 0 Architecture](assets/025/lineage-arch.png)
+
+Implementing this feature will require defining a Lienage API on the metadata service for Tables and Columns. When the API is called it will make calls to neo4j and whatever the source of lineage data is. An interface needs to be created to interact with an implementer's lineage service in a generic way. The data from the calls to these services will be put together to form the lineage response as defined below.
```

Comment:
Would this generic API call be made to any of the metadata proxies? or is this simply a configurable function which users will be able to extend themselves? 
Suppose a user has a proxy set to `Atlas`. For Stage 0, will this endpoint call a function inside the atlas_proxy?

Reply:
Yeah on the Base Proxy class I added a get_lineage method that can be implemented and will be called for any proxy when the endpoint is hit https://github.com/amundsen-io/amundsenmetadatalibrary/blob/master/metadata_service/proxy/base_proxy.py#L160

Reply:
So the idea is that with this method any lineage provider (like Marquez, spline or open lineage) can be supported, right?

Reply:
@mgorsk1 I think this is just step 0 which we should bring the backend model (or I assume we will bring) with ingestion in step 1/2. The reason on why we are doing this is because Lyft still uses a 3rd party vendor lineage service . It is easier to for the implementation in these kinds of steps by focusing on FE first with backend proxy; then build a backend model/ ingestion with push mechanism in a subsequent step.

Reply:
Yeah we will hopefully add a backend model for lineage in the future so that we can query the db directly for lineage rather than making ad hoc calls to a provider from metadata. In that case we would have to add support for lineage extraction on databuilder.

Reply:
Clear, for me support for external service is more than fine, just wanted to make sure I got it right.